New attributes with GID ranges for specific namespace

stavamichal opened the pull request:
- new GIDRanges entityless attribute to set ranges of gids managed by
 system Perun and replace logic of min and max gid entityless attributes
- new GIDRanges virtual facility attribute to replace logic of min and
max gid virtual attributes defined for facility
- these attributes will be used in the future instead minGID and maxGID
attributes, because there is only one interval between min and max
gid and we need to choose some specific gids to manage them too
- method to check and transfer data from gid ranges attribute is
located in ModulsUtils, because we will need it for further work in
other funcionality working with GID ranges
- added tests for this method

licehammer commented on perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_entityless_attribute_def_def_namespace_GIDRanges.java:
```diff
@@ -0,0 +1,100 @@
+		attr.setFriendlyName("namespace_GIDRanges");
+		attr.setDisplayName("GID ranges in namespace");
+		attr.setType(List.class.getName());
+		attr.setDescription("Manageable GID ranges in namespace.");
```
Please add note to description how correct value look like (syntax).

licehammer commented on perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_virt_GIDRanges.java:
```diff
@@ -0,0 +1,96 @@
+		return attribute;
+	}
+
+	public boolean setAttributeValue(PerunSessionImpl sess, Facility facility, Attribute attribute) throws InternalErrorException, WrongReferenceAttributeValueException {
```
I would rather not implement this method. This can be called only by perun admin and I would recommend to require to set entityless attribute directly (to avoid possible mistake, when one could thing he only changes settings for this facility).

zlamalp replied:
We discussed this with @stavamichal and this implementation differ from module default also in original GID module. I believe it throws exception, if you are trying to set different value, than is already stored and return false otherwise. In both cases it doesn't store/change anything. We are not sure, if exception is not expected somewhere in peruns logic. Like some automatic process wants to set required attributes, it sets value inside the object and call setAttribute() method. If it would only return false, process could continue with wrong value in object causing problems to other attributes. Like this it kill the process. But this code might not be relevant anymore. What do you think @licehammer ?

licehammer replied:
You are right Pavel. I just saw the implementation of the method itself and I didn't check what is does exactly. My bad. This method can stay as it is.

To your question: This method doesn't need to be implemented, because when you don't have the "fill" method, no automatic process should try to set this attribute. On the other hand, user might do that for example by modifying some attributes from GUI and then select and save all (including this one). In that case might be handy to have this method implemented as it is now.

stavamichal commented:
I fixed description and reworked the commit to have method separate from code of this specific module (we will need it for further work with gid ranges in other functionality). I also added tests for this method. It is prepared for review.
